src/editions/GatedEditionCreator.sol

```diff
@@ -1,95 +1,114 @@
+
+    constructor(address _editionImpl, address _minter) {
+        if (_editionImpl == address(0) || _minter == address(0)) {
             revert NullAddress();
         }
-        editionCreator = ISingleEditionMintableCreator(_editionCreator);
-        minter = IGatedEditionMinter(_minter);
-        timeCop = TimeCop(_timeCop);
 
+        editionImpl = _editionImpl;
+        minter = IGatedEditionMinter(_minter);
         showtimeVerifier = IGatedEditionMinter(_minter).showtimeVerifier();
     }
 
     /// Creates a new edition contract as a factory with a deterministic address
-    /// Important: None of these fields (except the Url fields with the same hash) can be changed after calling
     /// @param name Name of the edition contract
     /// @param description Metadata: Description of the edition entry
     /// @param animationUrl Metadata: Animation url (optional) of the edition entry
     /// @param imageUrl Metadata: Image url (semi-required) of the edition entry
     /// @param editionSize Total size of the edition (number of possible editions)
     /// @param royaltyBPS BPS amount of royalty
-    /// @param claimWindowDurationSeconds How long after deployment the edition can be claimed, in seconds
+    /// @param mintPeriodSeconds How long after deployment the edition can be claimed, in seconds
     /// @param signedAttestation the attestation to verify along with a corresponding signature
     /// @dev we expect the signed attestation's context to correspond to this contract's address
     /// @dev we expect the signed attestation's beneficiary to be the edition's creator
-    /// @return the address of the created edition
+    /// @return edition the address of the created edition
     function createEdition(
-        // ISingleEditionMintableCreator parameters
-        string memory name,
-        string memory description,
-        string memory animationUrl,
-        string memory imageUrl,
+        string calldata name,
+        string calldata description,
+        string calldata animationUrl,
+        string calldata imageUrl,
         uint256 editionSize,
         uint256 royaltyBPS,
-        // additional parameters
-        uint256 claimWindowDurationSeconds,
+        uint256 mintPeriodSeconds,
+        string calldata externalUrl,
+        string calldata creatorName,
         SignedAttestation calldata signedAttestation
-    ) external returns (address) {
+    ) external returns (IEdition edition) {
+        if (mintPeriodSeconds == 0 || mintPeriodSeconds > MAX_DURATION_SECONDS) {
+            revert InvalidTimeLimit(mintPeriodSeconds);
+        }
+
         validateAttestation(signedAttestation);
+        address creator = signedAttestation.attestation.beneficiary;
+
+        bytes32 salt = keccak256(abi.encodePacked(creator, name, animationUrl, imageUrl));
+        edition = IEdition(ClonesUpgradeable.cloneDeterministic(editionImpl, salt));
```

Comment:
whoa, is this a gas optimized deploy?

Reply:
kind of -- we use to delegate this to the regular `EditionCreator` (previously the mainstream ZORA factory SingleEditionMintableCreator), but I want the salt computation to involve the actual creator address so I moved it up here

I think the commit message has more deets

Reply:
see https://github.com/showtime-xyz/showtime-contracts/pull/3/commits/868393c1895423f1d718ca1dd07faec0450fb0e3

Reply:
ahh ok that makes sense! thank you!

Reply:
but as an added bonus, it should be a little more gas efficient (one less hop!)